tests/overlay-local-store/common.sh

```diff
@@ -8,38 +8,52 @@ requireEnvironment () {
 }
 
 setupConfig () {
-  echo "drop-supplementary-groups = false" >> "$NIX_CONF_DIR"/nix.conf
+  echo "require-drop-supplementary-groups = false" >> "$NIX_CONF_DIR"/nix.conf
   echo "build-users-group = " >> "$NIX_CONF_DIR"/nix.conf
 }
 
+
+
 storeDirs () {
-  storeA="$TEST_ROOT/store-a"
-  storeBTop="$TEST_ROOT/store-b"
-  storeB="local-overlay?root=$TEST_ROOT/merged-store&lower-store=$storeA&upper-layer=$storeBTop"
+  # Attempt to create store dirs on tmpfs volume.
+  # This ensures lowerdir, upperdir and workdir will be on
+  # a consistent filesystem that fully supports OverlayFS.
+  storeVolume="$TEST_ROOT/stores"
+  mkdir -p "$storeVolume"
+  mount -t tmpfs tmpfs "$storeVolume" || true  # But continue anyway if that fails.
```

Comment:
I'm unsure if this is strictly needed for the CI machines, but without this `nix-build` fails on my machine:
```
[197084.546631] overlayfs: upper fs does not support RENAME_WHITEOUT.                                                                                                                                 
[197084.546634] overlayfs: failed to set xattr on upper                                                                                                                                               
[197084.546635] overlayfs: ...falling back to xino=off.                                                                                                                                               
[197084.546635] overlayfs: upper fs missing required features.
```